pilot/pkg/model/push_context.go

```diff
@@ -1608,30 +1608,30 @@ func (ps *PushContext) mergeGateways(proxy *Proxy) *MergedGateway {
 
 // pre computes gateways for each network
 func (ps *PushContext) initMeshNetworks() {
-	if ps.Networks == nil || len(ps.Networks.Networks) == 0 {
-		return
-	}
-
 	ps.networkGateways = map[string][]*Gateway{}
-	for network, networkConf := range ps.Networks.Networks {
-		gws := networkConf.Gateways
-		if len(gws) == 0 {
-			// all endpoints in this network are reachable directly from others. nothing to do.
-			continue
-		}
 
-		registryNames := getNetworkRegistries(networkConf)
```

Comment:
What happened with this call ? Adding a new mechanism is good, but we can't just remove the old one, would break upgrades - we need to deprecate, warn - and remove in a future release.

Reply:
It's the same mechanism, same logic but pushed down into the service registry. Not sure if that's the right move after some thought - it might be better to have model.Service track the gateway and index Services that are gateways.